src/types/st-object.ts

```diff
@@ -1,11 +1,68 @@
 import { Field, FieldInstance } from "../enums";
 import { SerializedType, JsonObject } from "./serialized-type";
+import {
+  xAddressToClassicAddress,
+  isValidXAddress,
+} from "ripple-address-codec";
 import { BinaryParser } from "../serdes/binary-parser";
 import { BinarySerializer, BytesList } from "../serdes/binary-serializer";
 
 const OBJECT_END_MARKER = Buffer.from([0xe1]);
 const OBJECT_END_MARKER_NAME = "ObjectEndMarker";
 const OBJECT_FIELD_TYPE_NAME = "STObject";
+const ACCOUNT_ID_TYPE_NAME = "AccountID";
+const DESTINATION_FIELD_NAME = "Destination";
+const ACCOUNT_FIELD_NAME = "Account";
+const SOURCE_TAG_FIELD_NAME = "SourceTag";
+const DESTINATION_TAG_FIELD_NAME = "DestinationTag";
+
+/**
+ * Break down an X-Address into an account and a tag
+ *
+ * @param field Name of field
+ * @param xAddress X-Address cooresponding to field f
+ */
+function handleXAddress(field: string, xAddress: string): JsonObject {
+  if (!isValidXAddress(xAddress)) {
+    throw new Error(`Invalid X address ${xAddress}`);
+  }
+
+  const decoded = xAddressToClassicAddress(xAddress);
+
+  let tagName;
+  if (field === DESTINATION_FIELD_NAME) tagName = DESTINATION_TAG_FIELD_NAME;
+  else if (field === ACCOUNT_FIELD_NAME) tagName = SOURCE_TAG_FIELD_NAME;
+  else if (decoded.tag !== false)
+    throw new Error(`${field} cannot have an associated tag`);
+
+  return decoded.tag
+    ? { [field]: decoded.classicAddress, [tagName]: decoded.tag }
+    : { [field]: decoded.classicAddress };
+}
+
+/**
+ * Validate that two objects don't both have the same tag fields
```

Comment:
Why is this needed?

Reply:
Validating that the JSON object doesn't have both an X-Address and a SourceTag. What I probably should do instead is pass it in as a parameter to the `parseXAddress`  function.